components/shopify/sources/new-cancelled-order/new-cancelled-order.js

```diff
@@ -0,0 +1,40 @@
+const shopify = require("../../shopify.app.js");
+
+module.exports = {
+  key: "shopify-new-cancelled-order",
+  name: "New Cancelled Order",
+  description: "Emits an event each time a new order is cancelled.",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+    shopify,
+  },
+  async run() {
+    const lastUpdatedAt = this.db.get("last_updated_at") || null;
```

Comment:
We're trying to move towards a getter/setter approach when dealing with `$.service.db` to avoid potential typos and leverage encapsulation:

```js
module.exports = {
  methods: {
    _getLastUpdatedDate() {
      return this.db.get("last_updated_at") || null;
    },
    _setLastUpdatedDate(date) {
      this.db.set("last_updated_at", date);
    },
  },
  async run() {
    const lastUpdatedAt = this._getLastUpdatedDate();

    // Do some stuff...

    this._setLastUpdatedDate(results[results.length - 1].updated_at);
  },
};
```